oauth2client/client.py

```diff
@@ -837,6 +838,9 @@ def _generate_refresh_request_headers(self):
         if self.user_agent is not None:
             headers['user-agent'] = self.user_agent
 
+        if _METADATA_V1_URL_PREFIX in self.token_uri:
+            headers[_METADATA_FLAVOR_HEADER] = _DESIRED_METADATA_FLAVOR
```

Comment:
There is no other way on the `oauth2client.client.OAuth2Credentials` class, but this header should only be relevant to the `oauth2client.client.contrib.gce.AppAssertionCredentials` class. You should just over-ride `_generate_refresh_request_headers` there.